pytket_qir/gatesets/pyqir/pyqir.py

```diff
@@ -12,41 +12,67 @@
+    OpType.H: QirGate(
+        func_nat=FuncNat.QIS, func_name=FuncName.H, func_spec=FuncSpec.BODY
+    ),
+    OpType.X: QirGate(
+        func_nat=FuncNat.QIS, func_name=FuncName.X, func_spec=FuncSpec.BODY
+    ),
+    OpType.Y: QirGate(
+        func_nat=FuncNat.QIS, func_name=FuncName.Y, func_spec=FuncSpec.BODY
+    ),
+    OpType.Z: QirGate(
+        func_nat=FuncNat.QIS, func_name=FuncName.Z, func_spec=FuncSpec.BODY
+    ),
+    OpType.S: QirGate(
+        func_nat=FuncNat.QIS, func_name=FuncName.S, func_spec=FuncSpec.BODY
+    ),
+    OpType.Sdg: QirGate(
+        func_nat=FuncNat.QIS, func_name=FuncName.S, func_spec=FuncSpec.ADJ
+    ),
+    OpType.T: QirGate(
+        func_nat=FuncNat.QIS, func_name=FuncName.T, func_spec=FuncSpec.BODY
+    ),
+    OpType.Tdg: QirGate(
+        func_nat=FuncNat.QIS, func_name=FuncName.T, func_spec=FuncSpec.ADJ
+    ),
     OpType.Reset: QirGate(
-        opnat=FuncNat.QIS, opname=FuncName.RESET, opspec=FuncSpec.BODY
+        func_nat=FuncNat.QIS, func_name=FuncName.RESET, func_spec=FuncSpec.BODY
+    ),
+    OpType.CX: QirGate(
+        func_nat=FuncNat.QIS, func_name=FuncName.CX, func_spec=FuncSpec.BODY
+    ),
+    OpType.CZ: QirGate(
+        func_nat=FuncNat.QIS, func_name=FuncName.CZ, func_spec=FuncSpec.BODY
     ),
-    OpType.CX: QirGate(opnat=FuncNat.QIS, opname=FuncName.CX, opspec=FuncSpec.BODY),
-    OpType.CZ: QirGate(opnat=FuncNat.QIS, opname=FuncName.CZ, opspec=FuncSpec.BODY),
     OpType.Measure: QirGate(
-        opnat=FuncNat.QIS, opname=FuncName.MEASUREZ, opspec=FuncSpec.BODY
+        func_nat=FuncNat.QIS, func_name=FuncName.MEASUREZ, func_spec=FuncSpec.BODY
+    ),
+    OpType.Rx: QirGate(
+        func_nat=FuncNat.QIS, func_name=FuncName.Rx, func_spec=FuncSpec.BODY
+    ),
+    OpType.Ry: QirGate(
+        func_nat=FuncNat.QIS, func_name=FuncName.Ry, func_spec=FuncSpec.BODY
+    ),
+    OpType.Rz: QirGate(
+        func_nat=FuncNat.QIS, func_name=FuncName.Rz, func_spec=FuncSpec.BODY
     ),
-    OpType.Rx: QirGate(opnat=FuncNat.QIS, opname=FuncName.Rx, opspec=FuncSpec.BODY),
-    OpType.Ry: QirGate(opnat=FuncNat.QIS, opname=FuncName.Ry, opspec=FuncSpec.BODY),
-    OpType.Rz: QirGate(opnat=FuncNat.QIS, opname=FuncName.Rz, opspec=FuncSpec.BODY),
     OpType.CopyBits: QirGate(
-        opnat=FuncNat.QIS, opname=FuncName.READ_RES, opspec=FuncSpec.BODY
+        func_nat=FuncNat.QIS, func_name=FuncName.READ_RES, func_spec=FuncSpec.BODY
     ),
 }
 
 
 _PYQIR_TO_TK = {v: k for k, v in _TK_TO_PYQIR.items()}
 # Resolve some naming ambiguities.
 _PYQIR_TO_TK[
-    QirGate(opnat=FuncNat.QIS, opname=FuncName.CNOT, opspec=FuncSpec.BODY)
+    QirGate(func_nat=FuncNat.QIS, func_name=FuncName.CNOT, func_spec=FuncSpec.BODY)
 ] = OpType.CX
 
 
 PYQIR_GATES = CustomGateSet(
     name="PyQir",
-    template=Template("__quantum__${opnat}__${opname}__${opspec}"),
+    template=Template("__quantum__${func_nat}__${func_name}__${func_spec}"),
```

Comment:
Because it is a generic template for all function names. This is a convention from the QIR specs. Maybe a specialised version of this would work ? However, this would defeat the point in having a template though.